src/ispyb/__init__.py

```diff
@@ -1,44 +1,59 @@
 import configparser
 import logging
+import os
 
 __version__ = "6.0.1"
 
 _log = logging.getLogger("ispyb")
 
 
-def open(configuration_file):
-    """Create an ISPyB connection using settings from a configuration file.
-    This can be used either as a function call or as a context manager.
+def open(credentials=None):
+    """Create an ISPyB connection.
 
-    :param configuration_file: Full path to a file containing database
-                               credentials
-    :return: ISPyB connection object
+    Args:
+        credentials: a config file containing database credentials.
+            If `credentials=None` then look for a credentials file in the
+            "ISPYB_CREDENTIALS" environment variable.
+
+            Example credentials file::
+
+                [ispyb_mariadb_sp]
+                user = ispyb_api
+                pw = password_1234
+                host = localhost
+                port = 3306
+                db = ispybtest
+                reconn_attempts = 6
+                reconn_delay = 1
```

Comment:
Would it be better to refer to the example file rather than having the copy of half a file here?

Reply:
I thought it was more helpful to include the example here directly, rather than "I could tell you here, but instead I'm going to tell you to go and look in another file to find the answer".